// multi-ini/lib/filters.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.lowercase = lowercase;
exports.uppercase = uppercase;
exports.trim = trim;
exports.constants = constants;
exports.boolean = boolean;
exports.integer = integer;
var _ = require('lodash');

function lowercase(value) {
    return _.isString(value) ? value.toLowerCase() : value;
}

function uppercase(value) {
    return _.isString(value) ? value.toUpperCase() : value;
}

function trim(value) {
    return _.isString(value) ? value.trim() : value;
}

function constants(value) {
    var options = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

    if (!_.isString(value) || _.isEmpty(options.constants)) {
        return value;
    }

    _.forIn(options.constants, function (replacement, constant) {
        var matcher = new RegExp('" ' + constant + ' "', 'g');
        value = value.replace(matcher, '' + replacement);

        matcher = new RegExp('" ' + constant + '$', 'g');
        value = value.replace(matcher, replacement + '"');

        matcher = new RegExp('^' + constant + ' "', 'g');
        value = value.replace(matcher, '"' + replacement);
    });

    return value;
}

function boolean(value) {
    if (!_.isString(value) || value.length > 5) {
        return value;
    }

    switch (value.toLowerCase()) {
        case 'on':
        case 'yes':
        case 'true':
            return true;
        case 'off':
        case 'no':
        case 'none':
        case 'false':
            return false;
        default:
            return value;
    }
}

function integer(value) {
    var options = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};

    if (isNaN(value) || isNaN(parseInt(value, 10)) || options.keep_zero_prefix && value.length > 1 && value[0] === '0' // dont drop zero prefixes
    ) {
            return value;
        }

    return parseInt(value, 10);
}
